sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -80,30 +91,20 @@ const isNotInstalledError = (err: Error) => err.message.match(powerShellErrors.i
  * `Connect-AzAccount` from the command line.
  */
 export class AzurePowerShellCredential implements TokenCredential {
-  private commandStack: string[] = [formatCommand("pwsh")];
-
-  /**
-   * Creates an AzurePowerShellCredential, which will authenticate against
-   * Azure Active Directory using an existing login session created using
-   * `Connect-AzAccount` in PowerShell.
-   */
-  constructor() {
-    if (isWindows) {
-      this.commandStack.push(formatCommand("powershell"));
-    }
-  }
-
   /**
    * Gets the access token from Azure PowerShell
    * @param resource - The resource to use when getting the token
    */
   private async getAzurePowerShellAccessToken(
     resource: string
   ): Promise<{ Token: string; ExpiresOn: string }> {
-    for (const powerShellCommand of this.commandStack) {
+    // Clone the stack to avoid mutating it while iterating
+    for (const powerShellCommand of [...commandStack]) {
```

Comment:
This approach works, but feels kinda weird, since we're mixing powershell detection with the actual execution of the commands we need to get the token.

Would it be too much work to refactor this slightly to first determine the correct command in a loop (and remember that for later) and then run the other commands outside of the loop? Basically ditch the need for `continue`, is the crux of it.

Reply:
We spoke offline, and we will take this as a refactor before this library is released as GA.

Reply:
Can you make an issue?

Reply:
I like the idea of the refactoring! I wonder how expensive it would be